src/rosette/grammar/language.rkt

```diff
@@ -106,25 +106,49 @@
       (match (length args)
         [0 (lifted-apply-0-args self args)]
         [1 (lifted-apply-1-arg  self args)]
-        [2 (lifted-apply-2-args self args)]
+        [2 (let ([selftype (variable-symbol self)]) 
+          #;(lifted-apply-2-args self args)
+          (cond
+            [(equal? '+ selftype) (lifted-apply-arith-args self args)]
+            [(equal? '- selftype) (lifted-apply-arith-args self args)]
+            [(equal? '* selftype) (lifted-apply-arith-args self args)]
+            [(equal? '< selftype) (lifted-apply-cmp-args self args)]
+            [(equal? '= selftype) (lifted-apply-cmp-args self args)]
+            [(equal? 'vector-increment! selftype) (lifted-apply-vecincdec-args self args)]
+            [(equal? 'vector-decrement! selftype) (lifted-apply-vecincdec-args self args)]
+            [(equal? 'vector-set! selftype) (lifted-apply-vecset-args self args)]
+            [(equal? 'vector-ref selftype) (lifted-apply-vecref-args self args)]
+            [(equal? 'enum-set-add! selftype) (lifted-apply-enum-set-modify-type-args self args)]
+            [(equal? 'enum-set-remove! selftype) (lifted-apply-enum-set-modify-type-args self args)]
+            [(equal? 'enum-set-contains? selftype) (lifted-apply-enum-set-contains?-type-args self args)]
+            [(equal? 'map-ref selftype) (lifted-apply-map-ref-type-args self args)]
+            [(equal? 'map-set! selftype) (lifted-apply-map-set!-type-args self args)]
+            [(equal? 'add-edge! selftype) (lifted-apply-graph-modify-type-args self args)]
+            [(equal? 'remove-edge! selftype) (lifted-apply-graph-modify-type-args self args)]
+            [(equal? 'has-edge? selftype) (lifted-apply-graph-has-edge?-type-args self args)]
+            [(equal? 'vertex-parent selftype) (lifted-apply-graph-get-set-type-args self args)]
+            [(equal? 'vertex-children selftype) (lifted-apply-graph-get-set-type-args self args)]
+            [else (lifted-apply-2-args self args)]))]
```

Comment:
Replace with `(internal-error (format "Unknown procedure: ~a" selftype))` so that we know if we've forgotten a case. Also add a comment that this will break once we have higher order functions.